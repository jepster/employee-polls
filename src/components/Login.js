import {connect} from "react-redux";
import {useNavigate} from "react-router-dom";
import {Button, Dropdown, DropdownItem, DropdownMenu, DropdownToggle} from 'reactstrap';
import {useEffect, useState} from "react";
import {setAuthedUser} from "../actions/authedUserAction";
import {handleInitialPollsData} from "../actions/handleInitialPollsData";

const Login = ({ dispatch }) => {
  useEffect(() => {
    dispatch(handleInitialPollsData());
  });

  const [state, setState] = useState({
    dropdownOpen: false,
    selectedUser: 'Select user',
    password: '',
    showPasswordWarning: false,
  });
  const navigate = useNavigate();
  const setUserSignedIn = (e) => {
    e.preventDefault();

    if (state.password !== '') {
      navigate("/dashboard");
      setState({...state, showPasswordWarning: false});
      dispatch(setAuthedUser(state.selectedUser));
    } else {
      setState({...state, showPasswordWarning: true});
    }
  };

  const toggle = (option) => {
    let selectedUser = 'Select user';
    if (option.target.innerText !== 'Select user') {
      selectedUser = option.target.innerText;
    }
    setState({dropdownOpen: !state.dropdownOpen, selectedUser: selectedUser});
  };

  const setPassword = (password) => {
    setState({...state, password: password});
  }

  return (
    <div className="container">
      <h2>Sign in</h2>

      {state.showPasswordWarning ? (
        <div className="alert alert-danger" role="alert">
          Please enter a password.
        </div>
      ) : ''}

      <form>
        <label>User</label>
        <div className="row">
          <Dropdown>
            <Dropdown isOpen={state.dropdownOpen} toggle={toggle}>
              <DropdownToggle caret>
                {state.selectedUser}
              </DropdownToggle>
              <DropdownMenu>
                <DropdownItem>Tyler McGinnis</DropdownItem>
                <DropdownItem>Mat Samis</DropdownItem>
                <DropdownItem>Sarah Edo</DropdownItem>
              </DropdownMenu>
            </Dropdown>
          </Dropdown>
        </div>
        <div className="row pt-3">
          <label>Password (insert any you like)</label>
        </div>
        <input type="password" name="password" autoComplete="new-password" onChange={setPassword} />
        <div className="row pt-3">
          <div className="col-2">
            <Button color="primary" onClick={setUserSignedIn}>Submit</Button>
          </div>
        </div>
      </form>
    </div>
  );
}

export default connect()(Login);
