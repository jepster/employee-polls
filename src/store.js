import {configureStore} from "@reduxjs/toolkit";
import logger from "./middleware/logger";
import userReducer from "./reducers/userReducer";
import questionReducer from "./reducers/questionReducer";
import authedUserReducer from "./slices/authedUserSlice";

export const store = configureStore({
  reducer: {
    authedUser: authedUserReducer,
    users: userReducer,
    questions: questionReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().prepend(logger)
})
