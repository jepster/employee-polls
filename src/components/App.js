import {Fragment, useEffect} from "react";
import {connect} from "react-redux";
import {Route, Routes} from "react-router-dom";
import Login from "./Login";
import Nav from "./Nav";
import Dashboard from "./Dashboard";
import 'bootstrap/dist/css/bootstrap.min.css';
import NewPoll from "./NewPoll";
import Poll from "./Poll";
import PrivateRoute from "./PrivateRoute";
import {handleInitialData} from "../actions/shared";

const App = ({dispatch, authedUser}) => {
  useEffect(() => {
    dispatch(handleInitialData());
  });

  return (
    <Fragment>
      <div className="container">

        {authedUser ? <Nav/> : ''}

        <Routes>
          <Route path="/login" exact element={<Login/>}/>
          <Route path="/" exact element={<PrivateRoute><Dashboard/></PrivateRoute>}/>
          <Route path="/new-poll" exact element={<PrivateRoute><NewPoll/></PrivateRoute>}/>
          <Route path="/poll/:id" exact element={<PrivateRoute><Poll/></PrivateRoute>}/>
        </Routes>
      </div>
    </Fragment>
  );
};

const mapStateToProps = ({authedUser}) => ({
  authedUser: authedUser,
});

export default connect(mapStateToProps)(App);
