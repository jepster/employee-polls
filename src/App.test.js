import {getByText, render, screen} from '@testing-library/react';
import App from './components/App';
import {BrowserRouter} from "react-router-dom";
import {store} from "./store";
import {Provider} from "react-redux";
import Login from "./components/Login";
import '@testing-library/jest-dom'

describe("App", () => {

  it("should render the component", () => {
    let component;
    component = render(
      <Provider store={store}>
        <BrowserRouter>
          <App/>
        </BrowserRouter>
      </Provider>
    );
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });

  it("check if we have login form", () => {
    let component;
    component = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login/>
        </BrowserRouter>
      </Provider>
    );
    expect(component).toBeDefined();
    expect(component.getByText('User')).toBeInTheDocument();
    expect(component.getByText('Password (insert any you like)')).toBeInTheDocument();
    expect(component).toMatchSnapshot();
  });

});
