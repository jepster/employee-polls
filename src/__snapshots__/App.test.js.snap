// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App check if we have login form 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <h2>
          Sign in
        </h2>
        <form>
          <label>
            User
          </label>
          <div
            class="row"
          >
            <div
              class="dropdown"
            >
              <div
                class="dropdown"
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="true"
                  class="dropdown-toggle btn btn-secondary"
                  type="button"
                >
                  Select user
                </button>
                <div
                  aria-hidden="true"
                  class="dropdown-menu"
                  role="menu"
                  tabindex="-1"
                >
                  <button
                    class="dropdown-item"
                    role="menuitem"
                    tabindex="0"
                    type="button"
                  >
                    Tyler McGinnis
                  </button>
                  <button
                    class="dropdown-item"
                    role="menuitem"
                    tabindex="0"
                    type="button"
                  >
                    Mat Samis
                  </button>
                  <button
                    class="dropdown-item"
                    role="menuitem"
                    tabindex="0"
                    type="button"
                  >
                    Sarah Edo
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="row pt-3"
          >
            <label>
              Password (insert any you like)
            </label>
          </div>
          <input
            autocomplete="new-password"
            name="password"
            type="password"
          />
          <div
            class="row pt-3"
          >
            <div
              class="col-2"
            >
              <button
                class="btn btn-primary"
                type="button"
              >
                Submit
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <h2>
        Sign in
      </h2>
      <form>
        <label>
          User
        </label>
        <div
          class="row"
        >
          <div
            class="dropdown"
          >
            <div
              class="dropdown"
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                class="dropdown-toggle btn btn-secondary"
                type="button"
              >
                Select user
              </button>
              <div
                aria-hidden="true"
                class="dropdown-menu"
                role="menu"
                tabindex="-1"
              >
                <button
                  class="dropdown-item"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  Tyler McGinnis
                </button>
                <button
                  class="dropdown-item"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  Mat Samis
                </button>
                <button
                  class="dropdown-item"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  Sarah Edo
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="row pt-3"
        >
          <label>
            Password (insert any you like)
          </label>
        </div>
        <input
          autocomplete="new-password"
          name="password"
          type="password"
        />
        <div
          class="row pt-3"
        >
          <div
            class="col-2"
          >
            <button
              class="btn btn-primary"
              type="button"
            >
              Submit
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <nav
          class="nav"
        >
          <ul>
            <li>
              <a
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                href="/"
              >
                Leaderboard
              </a>
            </li>
            <li>
              <a
                href="/new-poll"
              >
                New poll
              </a>
            </li>
          </ul>
        </nav>
        <div
          class="container"
        >
          <h2>
            Sign in
          </h2>
          <form>
            <label>
              User
            </label>
            <div
              class="row"
            >
              <div
                class="dropdown"
              >
                <div
                  class="dropdown"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="true"
                    class="dropdown-toggle btn btn-secondary"
                    type="button"
                  >
                    Select user
                  </button>
                  <div
                    aria-hidden="true"
                    class="dropdown-menu"
                    role="menu"
                    tabindex="-1"
                  >
                    <button
                      class="dropdown-item"
                      role="menuitem"
                      tabindex="0"
                      type="button"
                    >
                      Tyler McGinnis
                    </button>
                    <button
                      class="dropdown-item"
                      role="menuitem"
                      tabindex="0"
                      type="button"
                    >
                      Mat Samis
                    </button>
                    <button
                      class="dropdown-item"
                      role="menuitem"
                      tabindex="0"
                      type="button"
                    >
                      Sarah Edo
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="row pt-3"
            >
              <label>
                Password (insert any you like)
              </label>
            </div>
            <input
              autocomplete="new-password"
              name="password"
              type="password"
            />
            <div
              class="row pt-3"
            >
              <div
                class="col-2"
              >
                <button
                  class="btn btn-primary"
                  type="button"
                >
                  Submit
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <nav
        class="nav"
      >
        <ul>
          <li>
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/"
            >
              Leaderboard
            </a>
          </li>
          <li>
            <a
              href="/new-poll"
            >
              New poll
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="container"
      >
        <h2>
          Sign in
        </h2>
        <form>
          <label>
            User
          </label>
          <div
            class="row"
          >
            <div
              class="dropdown"
            >
              <div
                class="dropdown"
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="true"
                  class="dropdown-toggle btn btn-secondary"
                  type="button"
                >
                  Select user
                </button>
                <div
                  aria-hidden="true"
                  class="dropdown-menu"
                  role="menu"
                  tabindex="-1"
                >
                  <button
                    class="dropdown-item"
                    role="menuitem"
                    tabindex="0"
                    type="button"
                  >
                    Tyler McGinnis
                  </button>
                  <button
                    class="dropdown-item"
                    role="menuitem"
                    tabindex="0"
                    type="button"
                  >
                    Mat Samis
                  </button>
                  <button
                    class="dropdown-item"
                    role="menuitem"
                    tabindex="0"
                    type="button"
                  >
                    Sarah Edo
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="row pt-3"
          >
            <label>
              Password (insert any you like)
            </label>
          </div>
          <input
            autocomplete="new-password"
            name="password"
            type="password"
          />
          <div
            class="row pt-3"
          >
            <div
              class="col-2"
            >
              <button
                class="btn btn-primary"
                type="button"
              >
                Submit
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
